src/de/jost_net/JVerein/server/FormularImpl.java

```diff
@@ -158,4 +158,94 @@ public Object getAttribute(String fieldName) throws RemoteException
     return super.getAttribute(fieldName);
   }
 
+  @Override
+  public int getZaehler() throws RemoteException
+  {
+    Integer counter = (Integer) getAttribute("zaehler");
+    return counter != null ? (int) counter : 0;
+  }
+
+  @Override
+  public void setZaehler(int zaehler) throws RemoteException
+  {
+    setAttribute("zaehler", zaehler);
+  }
+
+  @Override
+  public void setZaehlerToFormLink(int zaehler) throws RemoteException
+  {
+    DBIterator<Formular> formList = getLinked();
+    while (formList.hasNext())
+    {
+      Formular form = formList.next();
+      form.setZaehler(zaehler);
+      try
+      {
+        form.store();
+      }
+      catch (RemoteException e)
+      {
+        String fehler = "Fortlaufende Nr. kann nicht gespeichert werden. Siehe system log.";
+        Logger.error(fehler, e);
+        throw new RemoteException(fehler);
+      }
+      catch (ApplicationException e)
+      {
+        String fehler = "Fortlaufende Nr. kann nicht gespeichert werden. Siehe system log.";
+        Logger.error(fehler, e);
+      }
+    }
+  }
+
+  @Override
+  public Integer getFormLink() throws RemoteException
+  {
+    Long formId = (Long) getAttribute("formLink");
+    if (formId == null)
+    {
+      return 0;
+    }
+
+    return (Integer) Math.toIntExact((formId));
+  }
+
+  @Override
+  public void setFormLink(Integer formLink) throws RemoteException
+  {
+    setAttribute("formLink", formLink);
+  }
+
+  public DBIterator<Formular> getLinked()
```

Comment:
Es gibt zwei mögliche Implementierungen von getLinked: entweder alle Formular-IDs zurückgeben, die zusammengehören, oder alle außer der ID von this. Du gibst für this.getFormlink()>0 alle zurück, andernfalls alle außer this. Das sollte man nicht machen, weil es jeden Leser des Codes verwirrt. Außerdem wird für this.getFormLink()>0 der Zähler doppelt gespeichert. Du hast jetzt mehrfach ohne Antwort auf "resoved" gestellt. Du solltest entweder die Inkonsistenz beheben oder erklären, warum das eine gute Idee ist.

Reply:
Dein Codevorschlag habe ich getestet - es funktioniert leider nicht. Die Fallunterscheidung mache ich, damit die Dropdown-Box "Formularverknüpfung" im "Master-"Formular gesperrt wird, sobald mindestens ein anderes Formular auf den Master verlinkt (s. Screenshot ["Ein Formular auf das mindestens eine Verknüpfung liegt:"](https://github.com/openjverein/jverein/pull/72#issue-1977901101) ).

Reply:
Aber in dem Falll ändert sich doch gar nichts. Die angegebene Bedingung ist für den Fall getFormLink()>0, also nicht für den Master.

Reply:
Bitte selbst testen - [mein Branch feature/form-counter-and-linking](https://github.com/dippeal/jverein/tree/feature/form-counter-and-linking) ist public. Der aktuelle Code funktioniert. Codevorschläge schaue ich mir gerne an.